### ISSUE TITLE
queries are not extracted when `graphql` / `useStaticQuery` / `StaticQuery` are imported from `gatsby` using commonjs's `require` 

### ISSUE THREAD

**Issue by cedricdelpoux** [gatsbyjs/gatsby#28500]
## Description

Using the official `gatsby-plugin-layout` plugin and a `useStaticQuery` in the layout component throw the error:
`The result of this StaticQuery could not be fetched.`

### Steps to reproduce
```
module.exports = {
  siteMetadata: {
    title: "MyWebsite",
  },
  plugins: ["gatsby-plugin-layout"]
}
```
src/layouts/index.js
```
const {graphql, useStaticQuery} = require("gatsby")
const {Helmet} = require("react-helmet")
const React = require("react")

export default ({children, location, pageContext}) => {
  const query = graphql`
  const {
    site: {
      siteMetadata: {title},
    },
    image,
  } = useStaticQuery(graphql`
    query LayoutIndex {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)
  return (
    <>
      <Helmet>
        <title>{title}</title>
        <meta charSet="utf-8" />
        <meta property="og:site_name" content={title} />
        <meta property="og:type" content="website" />
      </Helmet>
      {children}
    </>
  )
}
```

### Expected result

Static Query should return data

### Actual result

 `WebpackError: The result of this StaticQuery could not be fetched.`

### Environment

  System:
    OS: macOS 11.0.1
    CPU: (4) x64 Intel(R) Core(TM) i5-4258U CPU @ 2.40GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.2.1 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.0.10 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 87.0.4280.88
    Firefox: 72.0.2
    Safari: 14.0.1
  npmPackages:
    gatsby: ^2.28.0 => 2.28.0 
    gatsby-plugin-layout: ^1.6.0 => 1.6.0

**Comment by pieh** [comment]
Hey @cedricdelpoux

It looks like our query extraction doesn't support commonjs style imports and needs ES imports

If you could try changing:
```diff
-const {graphql, useStaticQuery} = require("gatsby")
-const {Helmet} = require("react-helmet")
-const React = require("react")
+import { graphql, useStaticQuery } from "gatsby"
+import { Helmet } from "react-helmet"
+import * as React from "react"
```
and let me know if that fixes / workarounds the problem?

**Comment by cedricdelpoux** [comment]
If I use ES imports in my layout and I specify my component in my `gatsby-config.js`:
```
{
      resolve: "gatsby-plugin-layout",
      options: {
        component: require("./src/layouts/shell"),
      },
    },
```
I get the following error: `SyntaxError: Cannot use import statement outside a module`.

That is the only reason I used commonjs imports in this file. 

But if I don't pass my component as an option and use `gatsby-plugin-layout` with default options, rename my layout component to `src/layouts/index.js`, I can use ES imports.

It seems to work this way. But this is a workaround

**Comment by pieh** [comment]
```
{
      resolve: "gatsby-plugin-layout",
      options: {
        component: require("./src/layouts/shell"),
      },
    },
```
Do you actually use `require` here? This plugin expects relative path (a string) - see what is "default" when `component` is not specified - https://github.com/gatsbyjs/gatsby/blob/721f42aa210943a5cfe9b08fe93ad31ae315c447/packages/gatsby-plugin-layout/src/gatsby-node.js#L7-L14

I don't quite understand how it would even work (like ignoring static queries or modules error) if you actually `require` module in `gatsby-config`

**Comment by cedricdelpoux** [comment]
Nice catch! I made a PR to update the documentation about this.
Thank you

**Comment by pieh** [comment]
My initial comment was a bit wrong ;) I did mention relative path, but I did read the code a bit wrong. I mean - both should work due to our webpack config that actually does allow paths relative to project root directory, but current documention is also correct.

So `require.resolve(<path>)` as in documention is correct because it turns relative "import" to absolute path. but `require(<path>)` (so no `.resolve` part) is actually wrong because this returns a "module" and not a path to a module - that's why I did ask.

So this was just question about `require()` vs `require.resolve()` in your config as those are quite different
